spec/goldiloader/goldiloader_spec.rb

```diff
@@ -748,6 +748,185 @@
+            queries = { relation.klass => 1 }
+
+            if relation.options[:through]
+              through = items.first.association(relation.options[:through])
+              queries[through.klass] = 1
+            end
+
+            expect do
+              # Force load via auto include
+              rel = items.to_a.first.association(association)
+              # handle single and multi relations
+              rel.load_target
+
+              # Check that associations are loaded
+              items.each do |blog|
+                expect(blog.association(association)).to be_loaded
+              end
+            end.to execute_queries(queries)
+          end
+        end
+      end
+
+      let(:blogs) { Blog.order(:name).to_a }
+      let(:posts) { Post.order(:title).to_a }
+      let(:users) { User.order(:name).to_a }
+      let(:posts_through_blogs) { blogs.flat_map(&:posts) }
+      let(:tags) { Tag.where('parent_id IS NOT NULL').order(:name).to_a }
+      let(:users_through_tags) { tags.map(&:owner).select { |o| o.is_a?(User) } }
+
+      it_behaves_like "it auto eager loads association", :blogs, :posts
+      it_behaves_like "it auto eager loads association", :blogs, :authors
+      it_behaves_like "it auto eager loads association", :blogs, :posts_overridden
+      it_behaves_like "it auto eager loads association", :posts, :blog
+      it_behaves_like "it auto eager loads association", :users, :address
+      it_behaves_like "it auto eager loads association", :posts_through_blogs, :author
+      it_behaves_like "it auto eager loads association", :tags, :parent
+      it_behaves_like "it auto eager loads association", :users_through_tags, :posts
+
+      it "auto eager loads associations when the model is loaded via find" do
+        Goldiloader.enabled do
+          blog = Blog.find(blog1.id)
+
+          blog.posts.to_a.first.author
+          blog.posts.each do |post|
+            expect(post.association(:author)).to be_loaded
+          end
+        end
+      end
+
+      it "doesn't auto eager loads associations outside the block" do
+        blog = nil
+        Goldiloader.enabled do
+          blog = Blog.find(blog1.id)
+        end
+
+        arr = blog.posts.to_a
+        arr.first.author
+        arr.drop(1).each do |post|
+          expect(post.association(:author)).not_to be_loaded
+        end
+      end
+
+      it "sets inverse associations properly" do
+        # Force the first blog's posts to load
+        Goldiloader.enabled do
+          blogs.first.posts_with_inverse_of.to_a
+
+          blogs.each do |blog|
+            expect(blog.association(:posts_with_inverse_of)).to be_loaded
+            blog.posts_with_inverse_of.each do |post|
+              expect(post.association(:blog)).to be_loaded
+              expect(post.blog.object_id).to eq(blog.object_id)
+            end
+          end
+        end
+      end
+
+      it "marks auto eager loaded models as read only when the association is read only" do
```

Comment:
This example and the next example seem to duplicate examples [elsewhere](https://github.com/salsify/goldiloader/blob/master/spec/goldiloader/goldiloader_spec.rb#L269) in this file and don't seem particular to this change. Are they necessary?

Reply:
Indeed, this section was born as a copy paste :grimacing: